### ISSUE TITLE
Is there some way to ensure data is flushed into storage?

### ISSUE THREAD

**Issue by littlesome** [rt2zz/redux-persist#252]
I' m saving user's login state using this library, when user press the logout button, my app do this:

1. dispatch an action to clear the login state in redux
2. wait for 200ms, then reload the app

wait for 200ms is not good, if there's a callback when pending operations are finished would be better

**Comment by rt2zz** [comment]
There is not presently a way to do this. a flush option has been discussed.

Is there a reason you want to reload the app as opposed to sending out some kind of `RESET` action that resets state to initial state?

**Comment by Phredward** [comment]
Hi.  I would like to restart my react-native app to apply a code-push update.  Right before I do this, I store state in redux, and it is not persisted (unless I wait, which is an acceptable workaround for now).

I would be happy to check a variable in redux, call a method that tells me if my store is persisted, or have an alternative restart method that redux provides that finishes syncing and then restarts.

This raises another concern, what prevents a user from killing the app quickly after state is updated (besides the limits of human finger speed)?

**Comment by goodhyun** [comment]
I'd also like to make sure that every bits are persisted when the app is shutting down.
How's the progress of #217 by @simonacca ?

**Comment by rt2zz** [comment]
in practice I have not observed unsaved state to be an issue. That said I do like the idea of having a `flush` method which write all keys simultaneously and returns a promise. As long as this could be implemented cleanly and without adding a many bytes of code.

**Comment by Klathmon** [comment]
It's not a long-term solution, but I've got a "hack" that can let you at the very least get confirmation that your store was persisted before doing anything that requires it (like a page refresh).

In my case I'm using `localforage` as my store, so I just "proxy" the `setItem` key on the localforage object to be my own function which I can then use to run whatever I want once the `setItem` in localforage is complete. Here's the snippet that hopefully explains what I'm doing better than my gibberish:

```
const localforageSetItem = localforage.setItem.bind(localforage)

localforage.setItem = (key, value, callback) => {
  return localforageSetItem(key, value, (...args) => {
    console.log('Done persisting!')
    callback(...args)
  })
}
```


I'm going to use it to fire custom events so I can attach a listener to get notified when a store is fully persisted and it's safe to move on. It's not a particularly good solution, but it will work until either a manual persist gets implemented, or someone comes up with something much less horrible.

**Comment by rt2zz** [comment]
Thinking about this more, I would be favor of a `flush` method on the persistor which immediately writes all pending storage operations and returns a promise.

**Comment by rt2zz** [comment]
just landed in v5 `redux-persist@5.0.0-beta.13`

`await persistor.flush()`

Feedback appreciated
